src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -218,7 +218,7 @@ private void RestorePackages()
 
             if (functionResult is Task)
             {
-                functionResult = await ((Task)functionResult).ContinueWith(t => GetTaskResult(t));
+                functionResult = await ((Task)functionResult).ContinueWith(t => GetTaskResult(t), TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
This is the key change I wanted to make as it is in the function invocation path and ends up scheduling a simple continuation that can run synchronously.